### ISSUE TITLE
The Ontology Dilemma

### ISSUE THREAD

**Issue by lorenzoromani1983** [v5analytics/visallo#841]
Good afternoon.

I am really finding it hard to understand how I can make my own ontology working on Visallo.
Visallo comes out with a very limited set of entities/relationships, so I need to add a lot of classes (Person, Company, etc), and new "edges" between entities/vertices.
The only guide i have been able to find is this link in the Visallo website:

https://www.visallo.com/blog/creating-ontology/

But it misses key informations for non-expert users like me.

I followed the instructions in the above mentioned link, but i got stuck at the end:

what does it mean "We can change all of these to our custom IRI if you would like. Open simple.owl in an editor and replace all http://visallo.org/minimal with http://yourdomain.com/simple."?

Do i have to upload something and insert the web link to a newly created file? What specific file? 

At the end, the above-mentioned link says that I have to insert the following lines in the visallo.properties file:

repository.ontology=org.visallo.vertexium.model.ontology.InMemoryOntologyRepository
repository.ontology.owl.simple.iri=http://yourdomain.com/simple
repository.ontology.owl.simple.file=/home/jferner/dev/ontology-simple/simple.owl

I did, but i did not insert the second line since i have not uploaded anything anywhere. I replaced the third line with my own location for "simple.owl", but Visallo does not build as it detect several errors. 

I cannot find a comprehensive how-to in order to create my classes/relationship owl file and make it work. How should i proceed?

Thanks and merry Christmas :)

**Comment by EvanOxfeld** [comment]
@lorenzoromani1983 Thanks for your questions. This IRI is some resource identifier, perhaps for your organization. The IRI does not require a file to be uploaded although it could point to your ontology if it's accessible on the web somewhere. Also, it's ok to leave the IRI as `http://visallo.org/minimal`.

If you're still having issues, make sure `repository.ontology.owl.simple.iri` matches the IRI in your OWL file.

**Comment by lorenzoromani1983** [comment]
Thanks for replying. Unfortunately that does not work on my system (Ubuntu). Here the steps i followed: 

1) modified minimal.owl (in /visallo/config/ontology-minimal) as suggested in the "Create an Ontology" page, without editing the IRI address http://visallo.org/minimal
2) moved the files contained in /visallo/config/ontology-minimal to new directory: /visallo/config/ontology-simple
3) renamed minimal.owl in simple.owl

4) edited visallo.properties file, which is in the same directory /visallo/config, adding these lines:

repository.ontology=org.visallo.vertexium.model.ontology.InMemoryOntologyRepository
repository.ontology.owl.simple.iri=http://visallo.org/minimal
repository.ontology.owl.simple.file=/home/lorenzo/visallo/config/ontology-simple/simple.owl

as you suggested i made sure that repository.ontology.owl.simple.iri reflects the IRI address specied in the new simple.owl file (where I did NOT modify the "original" IRIs). 

5) deleted the empty ontology-minimal directory

6) built visallo from terminal with standard command.

I expected to get new "organization" and "employee" entities with organization > employs > employee relationship, but Visallo got stuck even before building with a very long series of errors within terminal.

In the end, I had to reinstall Visallo from scratch in order to make it work: also restoring the original configuration with directory ontology-minimal, minimal.owl and the original visallo.properties file did not work.

I noticed a difference between the "Create an Ontology" file and my version of Visallo, as in my system i have more entities than shown in the "Create an Ontology" page, and seems that Visallo is getting those entities (location, contact, mail, etc) from other sources. But I am surely wrong.

Thanks for your help!

Lorenzo
